flow-server/src/main/java/com/vaadin/flow/component/internal/UIInternals.java

```diff
@@ -542,6 +543,10 @@ public ExecutionCanceler addJavaScriptInvocation(
             JavaScriptInvocation invocation) {
         session.checkHasLock();
         pendingJsInvocations.add(invocation);
+      invocation.getOwner().ifPresent(node -> invocation.getOwner().get()
+             .addDetachListener(() -> pendingJsInvocations
+                     .removeIf(pendingInvocation -> pendingInvocation
+                             .equals(invocation))));
```

Comment:
the `node` param in the lambda is already `invocation.getOwner().get()`
```suggestion
      invocation.getOwner().ifPresent(node -> 
              node.addDetachListener(() -> pendingJsInvocations
                     .removeIf(pendingInvocation -> pendingInvocation
                             .equals(invocation))));
```

Reply:
Good one, shorter code is usually better. The only reason I do not commit this is I want to have only one commit for this change, then another commit for the other CP-s, to not push like 5-10 commits to the branch. 

So I squash logically related commits together.